src/scene/particle-system/particle-emitter.js

```diff
@@ -879,8 +879,9 @@ class ParticleEmitter {
                 customFace: this.emitter.orientation !== PARTICLEORIENTATION_SCREEN
             });
             this.shader = shader;
+            return shader;
         };
-        this.material.updateShader();
+        this.material.getShaderVariant();
```

Comment:
presumably this call is meant to cache the shader up-front, but presumably it doesn't help because `getShaderVariant` will be called again by `ForwardRenderer` anyway. should the call be removed?

Reply:
actually, programLib will cache the shader, so this call does help.